### PR TITLE
Update Juneteenth with observed dates

### PR DESCRIPTION
**Pull request by tylershuster**

### CODE REVIEW

**Comment by stelgenhof** [comment]
Hi @tylershuster Thanks for the PR! Can you please add some unit tests? Please see also: https://github.com/azuyalabs/yasumi/blob/develop/CONTRIBUTING.md
Thanks!

**Comment by tylershuster** [comment]
Will do. Sorry about the commit history; I'll clean that up. I'm a little confused about the [existing unit tests](https://github.com/azuyalabs/yasumi/blob/develop/tests/USA/JuneteenthTest.php) because it seems to already test for the change I added! (minus the "(observed)" portion)

**Comment by tylershuster** [comment]
Please pardon my tardiness. Things seem to be in order, so please review.

**Comment by stelgenhof** [comment]
@tylershuster Reran the checks and there seem to be some code style issues.

**Comment by stelgenhof** [comment]
@tylershuster Can you check once more your PR as there some minor errors? I'd like to include this fix in the next release I am planning. Thanks!

**Comment by tylershuster** [comment]
@stelgenhof I think the style is fine now — PHPCS doesn't throw any errors.

File: src/Yasumi/Provider/USA.php [hunk 1]
```diff
@@ -161,9 +161,18 @@ private function calculateMemorialDay(): void
     private function calculateJuneteenth(): void
     {
         if ($this->year >= 2021) {
+            $date = new DateTime("$this->year-6-19", DateTimeZoneFactory::getDateTimeZone($this->timezone));
+            $label = 'Juneteenth';
+            if ('Sunday' === $date->format('l')) {
```
Comment (stelgenhof) [inline src/Yasumi/Provider/USA.php:166]: I'd recommend to use the 'w' formatter (A numeric representation of the day (0 for Sunday, 6 for Saturday)) instead, as the string representation of the week day is less certain to be stable (may have future changes in casing, naming, etc.)

File: src/Yasumi/Provider/USA.php [hunk 2]
```diff
@@ -161,9 +161,18 @@ private function calculateMemorialDay(): void
...
+                $date->add(DateInterval::createfromdatestring('+1 day'));
+                $label .= ' (observed)';
+            } elseif ('Saturday' === $date->format('l')) {
```
Comment (stelgenhof) [inline src/Yasumi/Provider/USA.php:169]: I'd recommend to use the 'w' formatter (A numeric representation of the day (0 for Sunday, 6 for Saturday)) instead, as the string representation of the week day is less certain to be stable (may have future changes in casing, naming, etc.)

File: src/Yasumi/Provider/USA.php [hunk 3]
```diff
@@ -161,9 +161,18 @@ private function calculateMemorialDay(): void
...
+                $date->add(DateInterval::createfromdatestring('-1 day'));
+                $label .= ' (observed)';
+            }
             $this->addHoliday(new Holiday('juneteenth', [
-                'en' => 'Juneteenth',
-            ], new DateTime("$this->year-6-19", DateTimeZoneFactory::getDateTimeZone($this->timezone)), $this->locale));
+            'en' => $label,
+          ], $date, $this->locale));
```
Comment (stelgenhof) [inline src/Yasumi/Provider/USA.php:175]: For holidays that are observed/substituted due to that they may fall in the weekend for example, you should use the `SubstituteHoliday` class.

**Review by stelgenhof — CHANGES_REQUESTED** [review]

**Comment by tylershuster** [comment]
Thanks for the feedback. I think those issues are cleaned up. I tried to pattern it after an existing substitute holiday

File: tests/USA/JuneteenthTest.php [hunk 4]
```diff
@@ -100,14 +100,17 @@ public function testJuneteenthBefore2021(): void
     /**
      * Tests translated name of the holiday defined in this test.
      *
+     * This does not use a random year because the name can vary between
+     * "Juneteenth" and "Juneteenth (observed)"
+     *
```
[the remark this answers is not in the document]
Comment (tylershuster) [reply tests/USA/JuneteenthTest.php:105]: I can do that. I don't see a test for this particular method for substituted holidays. I was just going to let it generate a random year now that it generates substitute holidays properly.

**Comment by tylershuster** [comment]
Failure seems...incorrect

**Review by stelgenhof — APPROVED** [review]

**Comment by stelgenhof** [comment]
The unsuccessful checks are not directly related to this PR. I will merge it and resolve these afterwards.

**Pull request merged**